### PR TITLE
GitRelease methods documentation

### PR DESCRIPTION
**Pull request by jglapa**
Hi, here's my small contribution to the documentation in relation to #819 

Few things:
* the `delete_release` method always returns `True` so not sure if the return type is truly valid - potentially a bug
* I've tried generating the documentation locally to check if formatting is ok but failed - is there a guide to it?   I followed the read the docs starter guide but that didn't seem to reflect the way it's done here

### CODE REVIEW

File: github/GitRelease.py [hunk 1]
```diff
@@ -166,13 +166,21 @@ def zipball_url(self):
         return self._zipball_url.value
 
     def delete_release(self):
+        """
+        :calls: `DELETE /repos/:owner/:repo/releases/:release_id <https://developer.github.com/v3/repos/releases/#delete-a-release>`_
+        :rtype: bool
```
Comment (sfdye) [inline github/GitRelease.py]: Actually we don't need to return anything here. You can remove the return and add a `:rtype: None`

Comment (jglapa) [reply github/GitRelease.py]: done here: 0b2a6c30cd34356b3

Comment (jglapa) [reply github/GitRelease.py]: so there is a test expecting this method to return `True`. Shall I remove it?

Comment (sfdye) [reply github/GitRelease.py]: Of course, otherwise the tests will fail.

Comment (jglapa) [reply github/GitRelease.py]: ok, it was more a question if the assertion or the whole test case should be removed. I did the former now: 438b10c5935550f3

File: github/GitRelease.py [hunk 2]
```diff
@@ -166,13 +166,20 @@ def zipball_url(self):
         return self._zipball_url.value
 
     def delete_release(self):
-        headers, data = self._requester.requestJsonAndCheck(
+        """
+        :calls: `DELETE /repos/:owner/:repo/releases/:release_id <https://developer.github.com/v3/repos/releases/#delete-a-release>`_
+        :rtype: None
+        """
+        self._requester.requestJsonAndCheck(
```
[the remark this answers is not in the document]
Comment (jglapa) [reply github/GitRelease.py]: they are unused, what's the point?

[the remark this answers is not in the document]
Comment (sfdye) [reply github/GitRelease.py]: Simply for consistency. If we are gonna remove them, we will do it at one go.

[the remark this answers is not in the document]
Comment (jglapa) [reply github/GitRelease.py]: revert here 24654976

**Comment by sfdye** [comment]
@jglapa Thank you very much!

**Pull request merged**